### PR TITLE
test: add checking for examples in docs, travis

### PR DESCRIPTION
**Pull request by ajeddeloh**
Extract and try all the examples in the docs as part of the test process
to ensure we don't include invalid examples.

Fixes https://github.com/coreos/fcct/issues/24
Fixes https://github.com/coreos/fcct/issues/23

### CODE REVIEW

File: test [hunk 1]
```diff
@@ -8,5 +8,21 @@ res=$(gofmt -d .)
...
+shopt -s nullglob
+echo "Checking docs"
+for doc in docs/*md
+do
+	echo "Checking $doc"
+	# split each doc into a bunch of tmpfiles then run ct on them
+	sed -n '/^```yaml fedora-coreos-config/,/^```/ p' < ${doc} | csplit - '/```yaml fedora-coreos-config/' '{*}' -z --prefix tmpconfig -q
+	for i in tmpconfig*
+	do
+		cat "$i" | tail -n +2 | head -n -1 | go run internal/main.go > /dev/null || cat -n "$i"
```
Comment (arithx) [inline test]: Should probably do a `--strict` run. Our docs shouldn't have any warnings.

File: test [hunk 2]
```diff
@@ -8,5 +8,21 @@ res=$(gofmt -d .)
...
+		rm "$i"
+	done
+done
+
+echo ok
```
Comment (arithx) [inline test]: This ends up printing regardless of if the docs have errors, is that intentional?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+language: go
+go:
+  - "1.12.x"
+
+install: echo nop
```
Comment (ashcrow) [inline .travis.yml]: Is this needed?

Comment (ajeddeloh) [reply .travis.yml]: Yes. Travis has default behavior to instead go deps that needs an env var set to use go modules correctly. I'd set the envvar but we end up doing what it is doing anyway in `./test`. So this both avoids the problem and makes the test run a little faster. I'll add a comment.

**Comment by ajeddeloh** [comment]
Fixed.

File: test [hunk 4]
```diff
@@ -8,5 +8,21 @@ res=$(gofmt -d .)
...
+echo "Checking docs"
+for doc in docs/*md
+do
+	echo "Checking $doc"
+	# split each doc into a bunch of tmpfiles then run ct on them
+	sed -n '/^```yaml fedora-coreos-config/,/^```/ p' < ${doc} | csplit - '/```yaml fedora-coreos-config/' '{*}' -z --prefix tmpconfig -q
+	for i in tmpconfig*
+	do
+		cat "$i" | tail -n +2 | head -n -1 | go run internal/main.go > /dev/null || cat -n "$i"
+		rm "$i"
```
Comment (arithx) [inline test]: These files are not deleted if there's an error, should probably add a trap to delete them on exit and cancel it out after the loop.

Comment (ajeddeloh) [reply test]: I think GH is having a problem. Look at my branch this PR is based on and it's fixed there.

**Comment by ajeddeloh** [comment]
Updated to use trap, GH seems to be working now.

**Pull request merged**